xbmc/filesystem/MusicDatabaseDirectory/DirectoryNodeBoxsetDiscSongs.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DirectoryNodeBoxsetDiscSongs.h"
+
+#include "QueryParams.h"
+#include "guilib/LocalizeStrings.h"
+#include "music/MusicDatabase.h"
+
+using namespace XFILE::MUSICDATABASEDIRECTORY;
+
+CDirectoryNodeBoxsetDiscSongs::CDirectoryNodeBoxsetDiscSongs(const std::string& strName, CDirectoryNode* pParent)
+  : CDirectoryNode(NODE_TYPE_BOXSET_DISC_SONGS, strName, pParent)
+{
+
+}
+
+NODE_TYPE CDirectoryNodeBoxsetDiscSongs::GetChildType() const
+{
+  return NODE_TYPE_BOXSET_DISC_SONGS;
+}
+std::string CDirectoryNodeBoxsetDiscSongs::GetLocalizedName() const
+{
+
+  if (GetID() == -1)
+    return g_localizeStrings.Get(15102); // All Albums
+  return "";
+}
+
+bool CDirectoryNodeBoxsetDiscSongs::GetContent(CFileItemList& items) const
+{
+  CMusicDatabase musicdatabase;
+  if (!musicdatabase.Open())
+    return false;
+
+  CQueryParams params;
+  CollectQueryParams(params);
+
+  bool bSuccess=musicdatabase.GetBoxsetDiscSongs(BuildPath(), items);
+
```

Comment:
What is this method and node doing that is different from `CDirectoryNodeSong`?

Reply:
Ah well, I guess I could ask you what's the difference between CDirectoryNodeSong and CDirectoryAlbumCompilationSongs because I don't think that is needed either.  You shouldn't need the "compilations" flag when getting the songs as all the albums in items are already compilations and thus returning the songs for each album is enough.  I've tested that funnily enough and it works fine.

Reply:
Yes exactly, CDirectoryAlbumCompilationSongs is historic and superseded. 
Understandably you followed how compilations nodes had been done for boxed sets, but it was an outdated approach.